src/map/Map.cpp

```diff
@@ -0,0 +1,210 @@
+#include "Map.h"
+
+using namespace std;
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: Map(string file)
+ * Parameters:
+ *      file - File name, must be a CSV File
+ * Description:
+ * Map Object Constructor, creates a map object using the map selected file name.
+ */
+Map::Map(string file):fname(file) {
+    memset(AIMap.data(), 0, AIMap.size() * sizeof(AIMap[0]));
+    memset(mapdata.data(), 0,  mapdata.size() * sizeof(mapdata[0]));
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: int loadFileData()
+ * Description:
+ * Read's the Map CSV file map data with the appropriate tile IDs
+ */
+int Map::loadFileData() {
+    // Open file stream for selected map file
+    ifstream file(fname.c_str());
+
+    if (file.fail()) {
+        printf("failed to open file\n");
+        return 0;
+    }
+
+    // Array of wallstart positions
+    array<MapPoint, MAX_WALLS> wallStart;
+    // Indices
+    int j = 0;
+    int i = 0;
+
+    // Character Variable
+    char ch;
+    while (file >> noskipws >> ch) {
+        switch (ch) {
+            case ',':
+                continue;
+            case '\n':
+                j = 0;
+                i++;
+                continue;
+            case WALL:              //Wall piece found, update AI Map
+                AIMap[i][j] = 1;
+                break;
+            case WALL_START:        //Label start of wall rectangle
+                wallStart[wallStartCount].x = j;
+                wallStart[wallStartCount].y = i;
+                wallStartCount++;
+                AIMap[i][j] = 1;
+                break;
+            // case CONCRETE_START:    //Start of concrete area
+            case ZOMBIE_SPAWN:      //Zombie spawn Points
+                zombieSpawn[zombieSpawnCount].x = j;
+                zombieSpawn[zombieSpawnCount].y = i;
+                zombieSpawnCount++;
+                break;
+            case SHOP_SPOT:         // Shop spot points
+                shops[shopCount].x = j;
+                shops[shopCount].y = i;
+                shopCount++;
+                break;
+            case BASE_START:        //Start of base area
+                base.x = j;
+                base.y = i;
+                break;
+        }
+        mapdata[i][j] = ch;               //Store map file value
+        j++;
+    }
+    genWalls(wallStart);
+    // printData();
+    return 1;
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: void genWalls(const struct MapPoint wstart[MAX_WALLS])
+ * Parameters:
+ *      wstart - Array of wall start MapPoints.
+ * Description:
+ * Read's the Map CSV file map data with the appropriate tile IDs
+ */
+void Map::genWalls(const array<MapPoint, MAX_WALLS> wallStart) {
+    int nextx;
+    int nexty;
+    int endx;
+    int endy;
+
+    for (int q = 0; q < wallStartCount; q++) {
+        int startx = wallStart[q].x;
+        int starty = wallStart[q].y;
```

Comment:
It looks like q is only ever used to assign values to startx and starty
As such, turn it into a foreach loop
`for (const auto& elem : wallStart) {`
Then instead of creating a temporary, just use `elem.x` and `elem,y` further down in the loop